grassroots-backend/src/phone-canvass/PhoneCanvass.service.ts

```diff
@@ -26,18 +27,62 @@ import {
   PendingCall,
 } from "grassroots-shared/PhoneCanvass/PhoneCanvassSyncData";
 import type { Express } from "express";
+import {
+  CallResult,
+  CallStatus,
+} from "grassroots-shared/dtos/PhoneCanvass/CallStatus.dto";
+import { PhoneCanvassScheduler } from "./Scheduler/PhoneCanvassScheduler.js";
+import { Call } from "./Scheduler/PhoneCanvassCall.js";
+import { mergeMap } from "rxjs";
+import { PhoneCanvassSchedulerFactory } from "./Scheduler/PhoneCanvassSchedulerFactory.js";
 
 @Injectable()
 export class PhoneCanvassService {
   repo: EntityRepository<PhoneCanvassEntity>;
+  callsBySid = new Map<string, Call>();
```

Comment:
I'm not seeing where calls are being removed from this map. I believe this would result in a memory leak. Same for `schedulers`.

The way we're doing things, it seems like calls need to stick around until the end of the phone canvass. Can we remove the phone canvass and associated calls when the phone canvass is done?